vignettes/communicate.Rmd

```diff
@@ -0,0 +1,349 @@
+To use the badges, first call `usethis::use_lifecycle()` to embed the badge images in your package (you only need to do this once), then use `lifecycle::badge()` to insert a badge:
+
+```{r, eval = FALSE}
+#' `r lifecycle::badge("experimental")
+#' `r lifecycle::badge("deprecated")
+#' `r lifecycle::badge("superseded")
+```
+
+Deprecated functions also need to advertise their status when run.
+lifecycle provides `deprecate_warn()` which takes three main arguments:
+
+-   The first argument, `when`, gives the version number when the deprecation occurred.
+
+-   The second argument, `what`, describes exactly what was deprecated.
+
+-   The third argument, `with`, provides the recommended alternative.
+
+We'll cover the details shortly, but here are a few sample uses:
+
+```{r}
+lifecycle::deprecate_warn("1.0.0", old_fun(), new_fun())
+lifecycle::deprecate_warn("1.0.0", fun(), testthat::fun())
+lifecycle::deprecate_warn("1.0.0", fun(old_arg), fun(new_arg))
+```
+
+(Note that the message includes the package name --- this is automatically discovered from the environment of the calling function so will not work unless the function is called from the package namespace.)
+
+The following sections describe how to use lifecycle badges and functions together to handle a variety of common development tasks.
+
+## Functions
+
+### Deprecate a function
+
+First, add a badge to the the `@description` block[^1].
+Briefly describe why the deprecation occurred and what to use instead.
+
+[^1]: We only use an explicit `@description` when the description will be multiple paragraphs, as in these examples.
+
+```{r}
+#' Add two numbers
+#' 
+#' @description
+#' `r lifecycle::badge("deprecated")
+#' 
+#' This function was deprecated because we realised that it's
+#' a special case a [sum()] function.
+```
+
+Next, update the examples to show how to convert from the old usage to the new usage:
+
+```{r}
+#' @examples 
+#' add_two(1, 2)
+#' # ->
+#' sum(1, 2)
+```
+
+Then add `@keywords internal` to remove the function from the documentation index.
+If you use pkgdown, also check that it's no longer listed in `_pkgdown.yml`.
+These changes make reduce the chance of new users coming across a deprecated function, but don't prevent those who already know about it from referring to the docs.
+
+```{r}
+#' @keywords internal
+```
+
+You're now done with the docs, and it's time to add a warning when the user calls your function.
+Do this by adding call to `deprecate_warn()` on the first line of the function:
+
+```{r}
+add_two <- function(x, y) {
+  lifecycle::deprecate_warn("1.0.0", add_two(), base::sum())
+  x + y
+}
+
+add_two(1, 2)
+```
+
+`deprecate_warn()` generates user friendly messages for two common deprecation alternatives:
+
+-   Function in same package: `lifecycle::deprecate_warn("1.0.0", "fun_old()", "fun_new()")`
+
+-   Function in another package: `lifecycle::deprecate_warn("1.0.0", "old()", "package::new()")`
```

Comment:
It would be nice for the vignettes to be internally consistent re: quoting `what` and `with`. Based on what I've seen elsewhere, I'm not expecting quotes here.